### PR TITLE
Document common practice for choosing a service name

### LINKED ISSUE #106: Document common practice for choosing a service name
In order to go from 50 to 5,000 services, we need to have and understand our system for picking a canonical service name.

If you look at the filenames in https://github.com/ambanum/CGUs/tree/master/services most of the names are spelt and capitalized:
1. the same way as on wikipedia
2. Pascal case, but abbreviations in all-caps
3. preceded by parent service if applicable
4. just the commonly known name, no suffixes like ".com" or "Inc."
5. with spaces and dots removed

Examples:
ask.fm (1) Ask.fm (2) Ask.FM (5) AskFM
https://www.apple.com/ios/app-store/ (1) App Store (3) Apple App Store (5) AppleAppStore
Facebook Payments Inc. (4) Facebook Payments (5) FacebookPayments
Foursquare City Guide (4) Foursquare
reddit.com (1) reddit (2) Reddit

Exceptions (fix these?):
* [ ] GooglePlayStore instead of GooglePlay
* [ ] LastFm instead of LastFM (inconsistent because we also have AskFM)
* [ ] deviantART instead of DeviantArt
* [ ] hi5 instead of Hi5

### PR DESCRIPTION
**Pull request by Ndpnt**
Closes #106.

### CODE REVIEW

**Comment by MattiSG** [comment]
@TomHouriezDGE @DamienSeuxADLC as new contributors, your review of this major update in the contributors' documentation would be very precious 🙂

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,123 +1,163 @@
...
+The only mandatory keys are `fetch` and `select` (except for PDF files, for which only `fetch` is needed). Let’s start by defining these keys!
 
-If you're in doubt, please list the potential synonyms you have considered.
+### `fetch`
 
-Note that the desire to multiply documents types is probably a symptom that needs to be broken down into several services, see [practical guidelines for more information](https://github.com/ambanum/CGUs/wiki/Practical-guidelines#provider-prefixing).
+This property should simply contain the URL at which the document you want to track can be downloaded. HTML and PDF files are supported.
 
-### Filters
+When multiple versions coexist, terms are only tracked in their English version and for the European (EEA) jurisdiction.
```
Comment (Ndpnt) [inline CONTRIBUTING.md:93]: > Maybe highlighting this in bold?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,123 +1,163 @@
...
+> For example, the following selector will select the content between the element targeted by the CSS selector `#privacy-eea`, including it, and the element targeted by the CSS selector `#privacy-row`, excluding it:
+>
+> ```json
+> {
+>   "startBefore": "#privacy-eea",
+>   "endBefore": "#privacy-row"
+> }
+> ```
+
+### `remove`
```
Comment (Ndpnt) [inline CONTRIBUTING.md:127]: Yesterday I realized that this part was missing. Happy to see it here 👍

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -146,50 +186,52 @@ export function removeImages(document, { select: selector }) {
...
+  },
+  …
 }
 ```
 
-### Selecting jurisdiction and language
-
-For now, when multiple versions coexist, terms are only tracked in their English version and for the European (EEA) jurisdiction.
-
+## Testing your declaration
```
Comment (Ndpnt) [inline CONTRIBUTING.md:229]: Also add doc for `npm run validate:modified` command

**Comment by Ndpnt** [comment]
Awesome doc improvement, so much clearer than before.  👏 
Thanks @MattiSG!

**Comment by MattiSG** [comment]
@LucasVerneyDGE @TomHouriezDGE @DamienSeuxADLC last call for feedback before merging :wink:

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,14 +1,76 @@
...
+> If you have a hard time finding the service name, check out the wiki for [practical guidelines to find the service name](https://github.com/ambanum/CGUs/wiki/Naming-services#service-name), and feel free to mention your uncertainties in the pull request! We will help you improve the service name if necessary 🙂
+
+### Service ID
+
+The service ID is exposed to developers. It should be easy to handle with scripts and other tools.
+
+- Non-ASCII characters are not supported. Service IDs are derived from the service name by normalising it into ASCII.
+  - _Example: `RTÉ` → `RTE`_.
+  - _Example: `historielærer.dk` → `historielaerer.dk`_.
+  - _Example: `туту.ру` → `tutu.ru`_.
```
Comment (LucasVerneyDGE) [inline CONTRIBUTING.md:41]: The illustration here is that `туту.ру` is in a different alphabet? It's obvious when copy/pasting it, but Github (or my browser) is rendering it in a not so obvious way.

image

Maybe we could find a service name with ideograms to be extra explicit here?

Comment (MattiSG) [reply CONTRIBUTING.md:41]: > The illustration here is that `туту.ру` is in a different alphabet?

Yes, it is in Cyrillic.

> Maybe we could find a service name with ideograms to be extra explicit here?

Good idea, but I am not familiar of the transliteration process for this case 😕

Comment (LucasVerneyDGE) [reply CONTRIBUTING.md:41]: Maybe "抖音短视频" / "Douyin" (Chinese name of the TikTok app)?

See the intro of https://fr.wikipedia.org/wiki/TikTok.

**Review by MattiSG — APPROVED** [review]

**Comment by MattiSG** [comment]
Closes #106.

**Pull request merged**